spark/src/main/java/org/apache/iceberg/spark/SparkFixupTimestampType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.FixupTypes;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.TypeUtil;
+import org.apache.iceberg.types.Types;
+
+/**
+ * By default Spark TimestampType should be converted to Types.TimestampType.withZone() iceberg type.
+ * But we also can convert TimestampType to Types.TimestampType.withoutZone() iceberg type
+ * by setting SparkUtil.STORE_TIMESTAMP_WITHOUT_TIMEZONE_SESSION_PROPERTY to 'true'
+ */
+class SparkFixupTimestampType extends FixupTypes {
+
+  private final boolean useTimestampWithoutZone;
+
+  private SparkFixupTimestampType(Schema referenceSchema, boolean useTimestampWithoutZone) {
```

Comment:
Do we need this class to be parameterized? Can't we just not use it if the property isn't set?

Reply:
ie in the parent "If useTimestampWIthoutTimeZone THEN use SparkFixupTimestampType else don't"

Reply:
Make sense, I will change the code

Reply:
actually do we need to gate this at all? We won't allow calling the conversion code on a TS so it may just be fine to always use this time conversion and just add it to SparkFixup?

Ie

If handleTimestampWithouSession is false, we should fail in IcebergSource right?

Reply:
We need this class because this TS conversion will be used only when we creating new table from spark, in this case we will be able to select which timestamp type will be used `Types.TimestampType.withZone()` or `Types.TimestampType.withoutZone()` by [READ_TIMESTAMP_AS_TIMESTAMP_WITHOUT_TIMEZONE](https://github.com/apache/iceberg/blob/f2fdf2a5f9ffc64b44b353e9fd7d09ed21f8c416/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java#L52) session property

Reply:
Let me provide some details why we need to have ability to select which timestamp type (`Types.TimestampType.withZone()` or `Types.TimestampType.withoutZone()`) will be used for **new** iceberg tables created from spark. 
We have 2 main engines: Trino and Spark. In Trino we uses only timestamp types (without timezone), so we need to have ability to create new iceberg tables from spark with TS without time zone, just to have ability to not change a lot of our existing ETL processes

Reply:
Ok so basically there is "Read and write to existing tables regardless of timezone type" vs "Create new tables using non-spark timezone type". Let's make sure we have tests for both parameters

Reply:
we have tests for [both parameters](https://github.com/apache/iceberg/pull/2757/files/f2fdf2a5f9ffc64b44b353e9fd7d09ed21f8c416#diff-5735c57982a67ce763ffa1c7831f6d107751c44e5be3b07f288bd32c02510240R134)